setup.cfg

```diff
@@ -0,0 +1,93 @@
+[metadata]
+url = https://github.com/microsoft/FLAML
+maintainer = Microsoft Corporation
+maintainer_email = hpo@microsoft.com
+name = FLAML
+version = attr: flaml.version.__version__
+description = A fast library for automated machine learning and tuning
+long_description = file:README.md
+long_description_content_type = text/markdown; charset=UTF-8
+license = MIT
+classifiers =
+    Programming Language :: Python :: 3
+    License :: OSI Approved :: MIT License
+    Operating System :: OS Independent
+
+[options]
+python_requires = >= 3.7
```

Comment:
Question: does this way of packaging support python 3.6? I wanted to allow installation in python 3.6 by changing `>=3.7` into `>=3.6`.

Reply:
I am almost completely positive that there shouldn't be a problem but I can't seem to find anything definitive trying to look it up. Would you like for me to add Python 3.6 to the CI matrix?

Reply:
I knew that CI would not pass in Python 3.6 because an optional dependency (seqeval) doesn't support 3.6 and the test under nlp/ fails. One workaround is to skip those tests in Python 3.6.

Reply:
Then just in terms of replacing setup.py with setup.cfg there shouldn't be a problem, but perhaps it's worth opening a seperate issue for adding support for Python 3.6 and updating this only after the CI is set up.

Reply:
From the error message, it doesn't look specific to Python 3.6. What command did you use?

Reply:
```bash
python3.6 -m venv venv
. venv/bin/activate
pip install .
```
I am using debian 11 which I only installed recently, and so I didn't have python3.6 prior to this and had to install it. I did test that installing another package is working as expected after this happened and did not run into issues.

Reply:
Does the same command work in a different version of Python?

Reply:
Yes, running `pip install .` from the root directory while a Python 3.9 virtual envoronment is activated works without a problem.

Reply:
Got it. Is there a way to allow Python 3.6 installation? Though we do not plan to support Python 3.6 in future, I hope to give Python 3.6 users a grace period to be able to install it before they finish the migration.

Reply:
@sonichi is seems this is a local issue with my Python 3.6 installation. 
I edited `Dockerile` to use the Python 3.6 container, rebuild the devcontainer, edited `setup.py` so that is has `python_requires=">=3.6"` (this is required by pip) and installed the package without a problem.
```bash
root ➜ /workspaces/FLAML (main ✗) $ python -V
Python 3.6.15
root ➜ /workspaces/FLAML (main ✗) $ pip install .
Processing /workspaces/FLAML
  Preparing metadata (setup.py) ... done
Requirement already satisfied: NumPy>=1.17.0rc1 in /usr/local/lib/python3.6/site-packages (from FLAML==1.0.8) (1.19.5)
Requirement already satisfied: lightgbm>=2.3.1 in /usr/local/lib/python3.6/site-packages (from FLAML==1.0.8) (3.3.2)
Requirement already satisfied: xgboost>=0.90 in /usr/local/lib/python3.6/site-packages (from FLAML==1.0.8) (1.3.3)
Requirement already satisfied: scipy>=1.4.1 in /usr/local/lib/python3.6/site-packages (from FLAML==1.0.8) (1.5.4)
Requirement already satisfied: pandas>=1.1.4 in /usr/local/lib/python3.6/site-packages (from FLAML==1.0.8) (1.1.5)
Requirement already satisfied: scikit-learn>=0.24 in /usr/local/lib/python3.6/site-packages (from FLAML==1.0.8) (0.24.2)
ERROR: Package 'flaml' requires a different Python: 3.6.15 not in '>=3.7'
```

`setup.py` fix...

```bash
root ➜ /workspaces/FLAML (main ✗) $ pip install .
Processing /workspaces/FLAML
  Preparing metadata (setup.py) ... done
Requirement already satisfied: NumPy>=1.17.0rc1 in /usr/local/lib/python3.6/site-packages (from FLAML==1.0.8) (1.19.5)
Requirement already satisfied: lightgbm>=2.3.1 in /usr/local/lib/python3.6/site-packages (from FLAML==1.0.8) (3.3.2)
Requirement already satisfied: xgboost>=0.90 in /usr/local/lib/python3.6/site-packages (from FLAML==1.0.8) (1.3.3)
Requirement already satisfied: scipy>=1.4.1 in /usr/local/lib/python3.6/site-packages (from FLAML==1.0.8) (1.5.4)
Requirement already satisfied: pandas>=1.1.4 in /usr/local/lib/python3.6/site-packages (from FLAML==1.0.8) (1.1.5)
Requirement already satisfied: scikit-learn>=0.24 in /usr/local/lib/python3.6/site-packages (from FLAML==1.0.8) (0.24.2)
Requirement already satisfied: wheel in /usr/local/lib/python3.6/site-packages (from lightgbm>=2.3.1->FLAML==1.0.8) (0.37.0)
Requirement already satisfied: pytz>=2017.2 in /usr/local/lib/python3.6/site-packages (from pandas>=1.1.4->FLAML==1.0.8) (2022.1)
Requirement already satisfied: python-dateutil>=2.7.3 in /usr/local/lib/python3.6/site-packages (from pandas>=1.1.4->FLAML==1.0.8) (2.8.2)
Requirement already satisfied: threadpoolctl>=2.0.0 in /usr/local/lib/python3.6/site-packages (from scikit-learn>=0.24->FLAML==1.0.8) (3.1.0)
Requirement already satisfied: joblib>=0.11 in /usr/local/lib/python3.6/site-packages (from scikit-learn>=0.24->FLAML==1.0.8) (1.1.0)
Requirement already satisfied: six>=1.5 in /usr/local/lib/python3.6/site-packages (from python-dateutil>=2.7.3->pandas>=1.1.4->FLAML==1.0.8) (1.16.0)
Building wheels for collected packages: FLAML
  Building wheel for FLAML (setup.py) ... done
  Created wheel for FLAML: filename=FLAML-1.0.8-py3-none-any.whl size=196668 sha256=d04eb50aed3c5105ce87cb74b1785e5f1946acef1a13ab48ff48b28a2b7cc104
  Stored in directory: /tmp/pip-ephem-wheel-cache-76dmwcea/wheels/c2/94/1b/bf23904ffa88170981324a1811866c86312f7dcaa72fdd16df
Successfully built FLAML
Installing collected packages: FLAML
  Attempting uninstall: FLAML
    Found existing installation: FLAML 1.0.7
    Uninstalling FLAML-1.0.7:
      Successfully uninstalled FLAML-1.0.7
Successfully installed FLAML-1.0.8
WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
```

Reply:
[Sorry if you already read my previous comment, I had a mistake, this is the correct status]
I rebuilt the devcontainer with Python 3.6 and could verify the installation is working properly with the original `setup.py` but raising the same exception I was getting before for `setup.cfg`. I tried looking up possible causes but unfortunately haven't managed to figure this out so far. Seems like collecting requirements for the wheel build isn't succeeding, but nothing seems to resolve it. I tried adding a minimal `setup.py` but that didn't work either. 
I guess the options are either to drop support for 3.6 or wait with this PR until that happens.

Reply:
Thanks. If there is no strong reason to change the packaging format now, perhaps we should wait with this PR. In the same time, let's watch the progress of users' migration.